src/core/internal/array/construction.d

```diff
@@ -162,17 +196,25 @@ void _d_arraysetctor(Tarr : T[], T)(scope Tarr p, scope ref T value) @trusted
     import core.internal.postblit : postblitRecurse;
     import core.stdc.string : memcpy;
     import core.internal.traits : Unqual;
+    import core.lifetime : emplace;
     size_t walker;
     auto element_size = T.sizeof;
 
     try
     {
         foreach (i; 0 .. p.length)
         {
-            auto elem = cast(Unqual!T*)&p[walker];
-            // Copy construction is defined as bit copy followed by postblit.
-            memcpy(elem, &value, element_size);
-            postblitRecurse(*elem);
+            static if (__traits(hasCopyConstructor, T))
+            {
+                emplace(&p[walker], value);
```

Comment:
I have an observation and a question regarding this `emplace`
If the `struct` in the `unittest` isn't static, so
```
int counter;
struct S { ... }
```
Then this will segfault without an error message.

If I replace the `emplace` line with
```
p[walker] = T(value);
```
the this will give the following compilation error
```
 Error: cannot access frame pointer of i20365.__unittest_L29_C7.S
```

What are your thoughts on this? Expected behaviour? Bug? Change `emplace` with `p[walker] = T(value);` ?

Reply:
Changing the code to `p[walker].__ctor(value)` fixed this